Implement batch runner for postgresql.

raol opened the pull request:
In general it's almost copy/paste of MySqlBatchRunner class.
Slight modifications:
- changed queries to postgresql syntax
  UPDATE <what to update>
  SET <expression>
  FROM ( <generated select query> )
- slighly modified regular expression with self updates.
  ColumAlias is optional group now. Alias is not generated every time.
- When in TransactionScope it means we have open transaction, really, it is.
  Postgre treats such transaction as distributed, that's why opening new transaction fails.

raol committed:
Implement batch runner for postgresql.

In general it's almost copy/paste of MySqlBatchRunner class.
Slight modifications:
- changed queries to postgresql syntax
  UPDATE <what to update>
  SET <expression>
  FROM ( <generated select query> )
- slighly modified regular expression with self updates.
  ColumAlias is optional group now. Alias is not generated every time.
- When in TransactionScope it means we have open transaction, really, it is.
  Postgre treats such transaction as distributed, that's why opening new transaction fails.

KivraS commented:
Is there any other Relevant Commit or Fork Concerning Postgre SQL?
I have downloaded this  Commit Source , Needed to Change syntax from 
(schema.table ) to ("schema"."table") and also ( columnName) to ("columnName") . 
But otherwise it worked fine for a batch update i needed. Too bad it didn't got merged into the master branch , since a Postgresql handler is missing.

raol committed:
Remove detecting outside transaction.

- Previously we detected outside transactions using System.Transaction, but recently
  we overcame the issue using specific configuration of the PgSql devart provider.
  It's redundant code now and should be removed to make everything consistent
  with the rest of the code.

raol commented:
@KivraS, I'm not sure I've got your point. You are saying that the generated query by PgSqlBatch runner is incorrect. Right?
It seems you're using TableNames and ColumnNames instead of tablenames, i.e. your schema is not lowercase. Right?
If so, I will make additional changes to handle such cases.

KivraS commented:
"dbo" is the Schema name, "Transport" is the Table name.
For example. Previously the Runner Generated this sql.

`select * from dbo.Transport where Id = 450 `

If you run this code on pg Admin on Postgresql 9.3 you get 

> ERROR:  relation "dbo.transport" does not exist
LINE 1: select * from dbo.Transport where Id = 450

Correct syntax to work is 

`select * from dbo."Transport" where "Id" = 450`

I have managed to use your PsqlBatchRunner by altering some lines.
For example

Line 122
`deleteCommand.CommandText = sqlBuilder.ToString().Replace("[","").Replace("]", "");`
To
`deleteCommand.CommandText = sqlBuilder.ToString().Replace("[","\"").Replace("]", "\"");`

And similarly for column names and Update queries.

raol commented:
@KivraS Okay. Clear. Will fix my code. Thanks.

raol committed:
Put schema and table name in quotes

- Table and column names that are not lowercase are caused the error. So it's necessary
  to put them in quotes.
  More info here
  http://www.postgresql.org/docs/current/interactive/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS
